evm/router/router.go

```diff
@@ -228,30 +223,22 @@ func (r *routerImpl) getConnectionForNamespace(ns string) (routerv1.MsgClient, e
 	addr := res.Address
 	conn, err := grpc.Dial(
 		addr,
-		grpc.WithTransportCredentials(r.creds),
+		grpc.WithUnaryInterceptor(r.clientCallInterceptor),
```

Comment:
can also use grpc.WithPerRPCCredentials here too. If you want it to only be injected for a specific rpc endpoint, you can inject `grpc.PerRPCCredentials` as the call option for that rpc endpoint.